geocodingv7/geocoding.go

```diff
@@ -0,0 +1,88 @@
+package geocodingv7
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// Geocoding allows forward geocode addresses and geo-positions.
+// See https://developer.here.com/documentation/geocoding-search-api/dev_guide/topics/quick-start.html#send-a-request
+// for details about other parameters.
+func (s *GeocodingService) Geocoding(
+	ctx context.Context,
+	req *GeocodingRequest,
+) (_ *GeocodingResponse, err error) {
+	u, err := s.URL.Parse("geocode")
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Q == nil && req.Address == nil {
+		return nil, fmt.Errorf("InvalidArgument, either Q or QQ must be provided")
+	}
+
+	values := make(url.Values)
+	if req.GeoPosition != nil {
+		values.Add("at", fmt.Sprintf("%v,%v", req.GeoPosition.Lat, req.GeoPosition.Long))
+	}
+	if req.Q != nil {
+		values.Add("q", *req.Q)
+	}
+	if req.Address != nil {
+		values.Add("qq", FormatQualifiedQuery(*req.Address))
+	}
+	if req.In != nil {
+		values.Add("in", *req.In)
+	}
+
+	r, err := s.Client.NewRequest(ctx, u, http.MethodGet, values.Encode(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create get request: %v", err)
+	}
+	var resp GeocodingResponse
+	if err := s.Client.Do(r, &resp); err != nil {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+// FormatQualifiedQuery Takes an address and formats to a qualified query.
+// Takes all key value pairs and separated by semicolon.
```

Comment:
```suggestion
// The resulting string will look like <key1>=<value1>;<key2>=<value2>;... where the keys are all fields of the address object and the values are their values.
```
If we think describing the output string is important perhaps this can be a bit clearer to the user of the function?